File: app/src/app/data-builder/user-data-builder.ts
import { User } from '../user';

export class UserDataBuilder {

    user = new User()

    constructor() {
        // Nothing to do
    }

    setUserId(user_id: string): UserDataBuilder {
        console.log(user_id);
        this.user.setUserId(user_id);
        return this;
    }

    setUserName(user_name: string): UserDataBuilder {
        this.user.setUserName(user_name);
        return this;
    }

    setThumbUrl(thumb_url: string): UserDataBuilder {
        this.user.setThumbUrl(thumb_url);
        return this;
    }


    setAccessToken(access_token: string): UserDataBuilder {
        this.user.setAccessToken(access_token);
        return this;
    }

    setHostConversations(conversations?: string): UserDataBuilder {
        this.user.setHostConversations(conversations);
        return this;
    }

    setGuestConversations(conversations?: string): UserDataBuilder {
        this.user.setGuestConversations(conversations);
        return this;
    }

    setUserProperties(user_properties?: string): UserDataBuilder {
        this.user.setUserProperties(user_properties);
        return this;
    }

    setPlans(plans?: string): UserDataBuilder {
        this.setPlans(plans);
        return this;
    }

    getResult(): User{
        return this.user;
    }

}

File: server/util/userdata_builder.py
from server.util.logger import Logger

from server.const.const import User


class UserDataBuilder():
    log = Logger("UserDataBuilder")

    data = {}

    def set_user_id(self, user_id):
        self.data[User.KEY_USER_ID] = user_id
        return self

    def set_user_name(self, user_name):
        self.data[User.KEY_USER_NAME] = user_name
        return self

    def set_thumb_url(self, thumb_url):
        self.data[User.KEY_THUMB_URL] = thumb_url
        return self

    def set_access_token(self, access_token):
        self.data[User.KEY_ACCESS_TOKEN] = access_token
        return self

    # def set_access_token_expire_time(self, access_token_expire_time):
    #     self.data[User.KEY_ACCESS_TOKEN_EXPIRE_TIME] = access_token_expire_time
    #     return self

    def set_convs_host(self, convs_host):
        self.data[User.KEY_CONVERSATIONS_HOST] = convs_host
        return self

    def set_convs_guest(self, convs_guest):
        self.data[User.KEY_CONVERSATIONS_GUEST] = convs_guest
        return self

    def set_user_properties(self, user_properties):
        self.data[User.KEY_USER_PROPERTIES] = user_properties
        return self

    def set_plans(self, plans):
        self.data[User.KEY_PLANS] = plans
        return self

    def set_psid(self, psid):
        self.data[User.KEY_PSID] = psid
        return self

    def get_result(self):
        return self.data
